### ISSUE TITLE
BlenderBIM standalone windows in space boundaries

### ISSUE THREAD

**Issue by brunopostle** [IfcOpenShell/IfcOpenShell#3995]
A boundary might consist of one or more windows that are not parented into wall openings, these should be considered when generating a space from walls, and when generating a boundary (since 55c4ce0 IfcWall, IfcColumn, IfcMember, IfcVirtualElement, and IfcPlate are considered).

A workaround is to change the class of the window to Curtain Wall (which is a subclass of wall) and use this to generate a boundary, but this loses any parametric geometry.

See #1676

**Comment by aothms** [comment]
Small note. Curtain wall is not a subtype of wall.

Sent from a mobile device, excuse my brevity. Kind regards, Thomas

Op za 11 nov. 2023 10:28 schreef Bruno Postle ***@***.***>:

**Comment by brunopostle** [comment]
Ah, the Space detection only considers these five classes, but boundaries can be generated using from anything (Window, Curtain Wall, Fastener etc..)

**Comment by CyrilWaechter** [comment]
Yes maybe we should add curtain wall there: https://github.com/IfcOpenShell/IfcOpenShell/blob/f478ec8d6b97b070923c188a003ad06b8e7bad9f/src/blenderbim/blenderbim/bim/module/boundary/operator.py#L581
About the structural elements I don't know. It is often standalone or like integrated into a wall. For finer energy analysis it would be relevant and maybe for other usage but at the moment I feel it would only add more bloat. I think at the moment we should let user add it manually if he wants to. Or we need a UI to let user choose which element should be detected.
